src/hooks/core/auth/useUserAuthHook.ts

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import {useRecoilState} from 'recoil';
+
+import profileAtom from '../../../atom/profileAtom';
+import {Context} from '../../../context';
+import {ProfileContext} from '../../../../types/context/profilecontext.types';
+
+const useUserAuthHook = () => {
+  const [profile] = React.useContext(Context).profile;
+  const profileContext = profile as ProfileContext;
+
+  const [authAtomState, setAuth] = useRecoilState(profileAtom);
+
+  const {anonProfileId, signedProfileId, token, anonymousToken} = authAtomState;
+
+  const resetAuth = () =>
+    setAuth({
+      signedProfileId: null,
+      anonProfileId: null,
+      token: null,
+      anonymousToken: null
+    });
+
+  return {
+    profile: profileContext.myProfile,
+    signedProfileId,
+    anonProfileId,
+    token,
+    anonymousToken,
+
+    resetAuth,
+    setAuth
+  };
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useUserAuthHook` function is missing dependencies for the `useRecoilState` hook. This could lead to unexpected behavior if the dependencies change during the lifecycle of the component. To fix this, you should pass the dependencies as an argument to the `useRecoilState` hook.

```diff
-  const [authAtomState, setAuth] = useRecoilState(profileAtom);
+  const [authAtomState, setAuth] = useRecoilState(profileAtom, [profile]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->